CGo fails to find `.h` files on linux

`WORKSPACE`:
```
new_http_archive(
    name = "pcre",
    sha256 = "ccdf7e788769838f8285b3ee672ed573358202305ee361cfec7a4a4fb005bbc7",
    url = "http://ftp.exim.org/pub/pcre/pcre-8.39.tar.gz",
    strip_prefix = "pcre-8.39",
    build_file = "pcre.BUILD",
    )

git_repository(
    name = "io_bazel_rules_go",
    remote = "https://github.com/bazelbuild/rules_go.git",
    commit = "6efffcfd01c25afe7ecf6150fe50c51aec160ee0", # v0.5.3 plus some
)

load("@io_bazel_rules_go//go:def.bzl", "go_repositories", "new_go_repository")

go_repositories()
```

`BUILD.bazel` (auto-generated via `bazel run //:gazelle`):
```
load("@io_bazel_rules_go//go:def.bzl", "go_library")

go_library(
    name = "go_default_library",
    srcs = ["pcre.go"],
    cgo = True,
    cdeps = ["@pcre//:pcre"],
    visibility = ["//visibility:public"],
)
```

Golang (`pcre.go`):
```go
/*
#include <pcre.h>
*/
import "C"
// Would add actual code, but this is enough to cause failure
```

This code builds fine running on OSX, but fails when run on linux (including inside a docker container) with:
```
ERROR: /project/pcre/BUILD.bazel:3:1: CGoCodeGen //pcre:go_default_library.cgo_codegen failed (Exit 1).
/root/.cache/bazel/_bazel_root/2ca1f4ebdc59348ffdc31d97a51a98d5/bazel-sandbox/2430527535312927236/execroot/com_returnpath_godzilla/pcre/pcre.go:59:18: fatal error: pcre.h: No such file or directory
 #include <pcre.h>
                  ^
compilation terminated.
CgoCodegen: error running cgo: exit status 1
```

Sure:
```
root@ce530054c3ef:/project/pcre# bazel build -s ...
INFO: Found 8 targets...
>>>>> # @pcre//:pcre_h [action 'Executing genrule @pcre//:pcre_h']
(cd /root/.cache/bazel/_bazel_root/2ca1f4ebdc59348ffdc31d97a51a98d5/execroot/test && \
  exec env - \
    PATH=/root/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin \
  /bin/bash -c 'source external/bazel_tools/tools/genrule/genrule-setup.sh; sed -e s/@PCRE_MAJOR@/8/    -e s/@PCRE_MINOR@/39/    -e s/@PCRE_PRERELEASE@//    -e s/@PCRE_DATE@/redacted/    external/pcre/pcre.h.in >bazel-out/local-fastbuild/genfiles/external/pcre/pcre.h')
>>>>> # @io_bazel_rules_go//go/tools/builders:cgo [action 'GoBuildTool external/io_bazel_rules_go/go/tools/builders/cgo [for host]']
(cd /root/.cache/bazel/_bazel_root/2ca1f4ebdc59348ffdc31d97a51a98d5/execroot/test && \
  exec env - \
    GOROOT=/root/.cache/bazel/_bazel_root/2ca1f4ebdc59348ffdc31d97a51a98d5/external/go1_8_3_linux_amd64 \
  external/go1_8_3_linux_amd64/bin/go build -o bazel-out/host/bin/external/io_bazel_rules_go/go/tools/builders/cgo external/io_bazel_rules_go/go/tools/builders/flags.go external/io_bazel_rules_go/go/tools/builders/filter.go external/io_bazel_rules_go/go/tools/builders/cgo.go)
>>>>> # @pcre//:pcre [action 'Compiling external/pcre/pcre_version.c']
(cd /root/.cache/bazel/_bazel_root/2ca1f4ebdc59348ffdc31d97a51a98d5/execroot/test && \
  exec env - \
    PATH=/root/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin \
    PWD=/proc/self/cwd \
  /usr/bin/gcc -U_FORTIFY_SOURCE -fstack-protector -Wall -B/usr/bin -B/usr/bin -Wunused-but-set-parameter -Wno-free-nonheap-object -fno-omit-frame-pointer -MD -MF bazel-out/local-fastbuild/bin/external/pcre/_objs/pcre/external/pcre/pcre_version.pic.d -fPIC -iquote external/pcre -iquote bazel-out/local-fastbuild/genfiles/external/pcre -iquote external/bazel_tools -iquote bazel-out/local-fastbuild/genfiles/external/bazel_tools -isystem external/pcre -isystem bazel-out/local-fastbuild/genfiles/external/pcre -isystem external/bazel_tools/tools/cpp/gcc3 '-DHAVE_BCOPY=1' '-DHAVE_INTTYPES_H=1' '-DHAVE_MEMMOVE=1' '-DHAVE_STDINT_H=1' '-DHAVE_STRERROR=1' '-DHAVE_SYS_STAT_H=1' '-DHAVE_SYS_TYPES_H=1' '-DHAVE_UNISTD_H=1' '-DLINK_SIZE=2' '-DMATCH_LIMIT=10000000' '-DMATCH_LIMIT_RECURSION=1000' '-DMAX_NAME_COUNT=10000' '-DMAX_NAME_SIZE=32' '-DNEWLINE=10' -DNO_RECURSE '-DPARENS_NEST_LIMIT=50' '-DPCRE_STATIC=1' '-DPOSIX_MALLOC_THRESHOLD=10' '-DSTDC_HEADERS=1' -DSUPPORT_UCP -DSUPPORT_UTF -fno-canonical-system-headers -Wno-builtin-macro-redefined '-D__DATE__="redacted"' '-D__TIMESTAMP__="redacted"' '-D__TIME__="redacted"' -c external/pcre/pcre_version.c -o bazel-out/local-fastbuild/bin/external/pcre/_objs/pcre/external/pcre/pcre_version.pic.o)
>>>>> # @pcre//:pcre [action 'Compiling external/pcre/pcre_jit_compile.c']
(cd /root/.cache/bazel/_bazel_root/2ca1f4ebdc59348ffdc31d97a51a98d5/execroot/test && \
  exec env - \
    PATH=/root/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin \
    PWD=/proc/self/cwd \
  /usr/bin/gcc -U_FORTIFY_SOURCE -fstack-protector -Wall -B/usr/bin -B/usr/bin -Wunused-but-set-parameter -Wno-free-nonheap-object -fno-omit-frame-pointer -MD -MF bazel-out/local-fastbuild/bin/external/pcre/_objs/pcre/external/pcre/pcre_jit_compile.pic.d -fPIC -iquote external/pcre -iquote bazel-out/local-fastbuild/genfiles/external/pcre -iquote external/bazel_tools -iquote bazel-out/local-fastbuild/genfiles/external/bazel_tools -isystem external/pcre -isystem bazel-out/local-fastbuild/genfiles/external/pcre -isystem external/bazel_tools/tools/cpp/gcc3 '-DHAVE_BCOPY=1' '-DHAVE_INTTYPES_H=1' '-DHAVE_MEMMOVE=1' '-DHAVE_STDINT_H=1' '-DHAVE_STRERROR=1' '-DHAVE_SYS_STAT_H=1' '-DHAVE_SYS_TYPES_H=1' '-DHAVE_UNISTD_H=1' '-DLINK_SIZE=2' '-DMATCH_LIMIT=10000000' '-DMATCH_LIMIT_RECURSION=1000' '-DMAX_NAME_COUNT=10000' '-DMAX_NAME_SIZE=32' '-DNEWLINE=10' -DNO_RECURSE '-DPARENS_NEST_LIMIT=50' '-DPCRE_STATIC=1' '-DPOSIX_MALLOC_THRESHOLD=10' '-DSTDC_HEADERS=1' -DSUPPORT_UCP -DSUPPORT_UTF -fno-canonical-system-headers -Wno-builtin-macro-redefined '-D__DATE__="redacted"' '-D__TIMESTAMP__="redacted"' '-D__TIME__="redacted"' -c external/pcre/pcre_jit_compile.c -o bazel-out/local-fastbuild/bin/external/pcre/_objs/pcre/external/pcre/pcre_jit_compile.pic.o)
>>>>> # @pcre//:pcre [action 'Compiling external/pcre/pcre_newline.c']
(cd /root/.cache/bazel/_bazel_root/2ca1f4ebdc59348ffdc31d97a51a98d5/execroot/test && \
  exec env - \
    PATH=/root/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin \
    PWD=/proc/self/cwd \
  /usr/bin/gcc -U_FORTIFY_SOURCE -fstack-protector -Wall -B/usr/bin -B/usr/bin -Wunused-but-set-parameter -Wno-free-nonheap-object -fno-omit-frame-pointer -MD -MF bazel-out/local-fastbuild/bin/external/pcre/_objs/pcre/external/pcre/pcre_newline.pic.d -fPIC -iquote external/pcre -iquote bazel-out/local-fastbuild/genfiles/external/pcre -iquote external/bazel_tools -iquote bazel-out/local-fastbuild/genfiles/external/bazel_tools -isystem external/pcre -isystem bazel-out/local-fastbuild/genfiles/external/pcre -isystem external/bazel_tools/tools/cpp/gcc3 '-DHAVE_BCOPY=1' '-DHAVE_INTTYPES_H=1' '-DHAVE_MEMMOVE=1' '-DHAVE_STDINT_H=1' '-DHAVE_STRERROR=1' '-DHAVE_SYS_STAT_H=1' '-DHAVE_SYS_TYPES_H=1' '-DHAVE_UNISTD_H=1' '-DLINK_SIZE=2' '-DMATCH_LIMIT=10000000' '-DMATCH_LIMIT_RECURSION=1000' '-DMAX_NAME_COUNT=10000' '-DMAX_NAME_SIZE=32' '-DNEWLINE=10' -DNO_RECURSE '-DPARENS_NEST_LIMIT=50' '-DPCRE_STATIC=1' '-DPOSIX_MALLOC_THRESHOLD=10' '-DSTDC_HEADERS=1' -DSUPPORT_UCP -DSUPPORT_UTF -fno-canonical-system-headers -Wno-builtin-macro-redefined '-D__DATE__="redacted"' '-D__TIMESTAMP__="redacted"' '-D__TIME__="redacted"' -c external/pcre/pcre_newline.c -o bazel-out/local-fastbuild/bin/external/pcre/_objs/pcre/external/pcre/pcre_newline.pic.o)
>>>>> # @pcre//:pcre [action 'Compiling external/pcre/pcre_study.c']
(cd /root/.cache/bazel/_bazel_root/2ca1f4ebdc59348ffdc31d97a51a98d5/execroot/test && \
  exec env - \
    PATH=/root/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin \
    PWD=/proc/self/cwd \
  /usr/bin/gcc -U_FORTIFY_SOURCE -fstack-protector -Wall -B/usr/bin -B/usr/bin -Wunused-but-set-parameter -Wno-free-nonheap-object -fno-omit-frame-pointer -MD -MF bazel-out/local-fastbuild/bin/external/pcre/_objs/pcre/external/pcre/pcre_study.pic.d -fPIC -iquote external/pcre -iquote bazel-out/local-fastbuild/genfiles/external/pcre -iquote external/bazel_tools -iquote bazel-out/local-fastbuild/genfiles/external/bazel_tools -isystem external/pcre -isystem bazel-out/local-fastbuild/genfiles/external/pcre -isystem external/bazel_tools/tools/cpp/gcc3 '-DHAVE_BCOPY=1' '-DHAVE_INTTYPES_H=1' '-DHAVE_MEMMOVE=1' '-DHAVE_STDINT_H=1' '-DHAVE_STRERROR=1' '-DHAVE_SYS_STAT_H=1' '-DHAVE_SYS_TYPES_H=1' '-DHAVE_UNISTD_H=1' '-DLINK_SIZE=2' '-DMATCH_LIMIT=10000000' '-DMATCH_LIMIT_RECURSION=1000' '-DMAX_NAME_COUNT=10000' '-DMAX_NAME_SIZE=32' '-DNEWLINE=10' -DNO_RECURSE '-DPARENS_NEST_LIMIT=50' '-DPCRE_STATIC=1' '-DPOSIX_MALLOC_THRESHOLD=10' '-DSTDC_HEADERS=1' -DSUPPORT_UCP -DSUPPORT_UTF -fno-canonical-system-headers -Wno-builtin-macro-redefined '-D__DATE__="redacted"' '-D__TIMESTAMP__="redacted"' '-D__TIME__="redacted"' -c external/pcre/pcre_study.c -o bazel-out/local-fastbuild/bin/external/pcre/_objs/pcre/external/pcre/pcre_study.pic.o)
>>>>> # @pcre//:pcre [action 'Compiling external/pcre/pcre_exec.c']
(cd /root/.cache/bazel/_bazel_root/2ca1f4ebdc59348ffdc31d97a51a98d5/execroot/test && \
  exec env - \
    PATH=/root/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin \
    PWD=/proc/self/cwd \
  /usr/bin/gcc -U_FORTIFY_SOURCE -fstack-protector -Wall -B/usr/bin -B/usr/bin -Wunused-but-set-parameter -Wno-free-nonheap-object -fno-omit-frame-pointer -MD -MF bazel-out/local-fastbuild/bin/external/pcre/_objs/pcre/external/pcre/pcre_exec.pic.d -fPIC -iquote external/pcre -iquote bazel-out/local-fastbuild/genfiles/external/pcre -iquote external/bazel_tools -iquote bazel-out/local-fastbuild/genfiles/external/bazel_tools -isystem external/pcre -isystem bazel-out/local-fastbuild/genfiles/external/pcre -isystem external/bazel_tools/tools/cpp/gcc3 '-DHAVE_BCOPY=1' '-DHAVE_INTTYPES_H=1' '-DHAVE_MEMMOVE=1' '-DHAVE_STDINT_H=1' '-DHAVE_STRERROR=1' '-DHAVE_SYS_STAT_H=1' '-DHAVE_SYS_TYPES_H=1' '-DHAVE_UNISTD_H=1' '-DLINK_SIZE=2' '-DMATCH_LIMIT=10000000' '-DMATCH_LIMIT_RECURSION=1000' '-DMAX_NAME_COUNT=10000' '-DMAX_NAME_SIZE=32' '-DNEWLINE=10' -DNO_RECURSE '-DPARENS_NEST_LIMIT=50' '-DPCRE_STATIC=1' '-DPOSIX_MALLOC_THRESHOLD=10' '-DSTDC_HEADERS=1' -DSUPPORT_UCP -DSUPPORT_UTF -fno-canonical-system-headers -Wno-builtin-macro-redefined '-D__DATE__="redacted"' '-D__TIMESTAMP__="redacted"' '-D__TIME__="redacted"' -c external/pcre/pcre_exec.c -o bazel-out/local-fastbuild/bin/external/pcre/_objs/pcre/external/pcre/pcre_exec.pic.o)
>>>>> # //pcre:go_default_library.cgo_codegen [action 'CGoCodeGen //pcre:go_default_library.cgo_codegen']
(cd /root/.cache/bazel/_bazel_root/2ca1f4ebdc59348ffdc31d97a51a98d5/execroot/test && \
  exec env - \
    CGO_LDFLAGS=bazel-out/local-fastbuild/bin/external/pcre/libpcre.lo \
    GOARCH=amd64 \
    GOOS=linux \
    GOROOT=/root/.cache/bazel/_bazel_root/2ca1f4ebdc59348ffdc31d97a51a98d5/external/go1_8_3_linux_amd64 \
  bazel-out/host/bin/external/io_bazel_rules_go/go/tools/builders/cgo external/go1_8_3_linux_amd64/bin/go -cc /usr/bin/gcc -objdir bazel-out/local-fastbuild/bin/pcre/go_default_library.cgo.dir -src 'bazel-out/local-fastbuild/bin/pcre/go_default_library.cgo.dir/pcre_pcre.cgo1.go=pcre/pcre.go' -- -- -I pcre -iquote '${execroot}/external/pcre' -iquote '${execroot}/bazel-out/local-fastbuild/genfiles/external/pcre' -iquote '${execroot}/external/bazel_tools' -iquote '${execroot}/bazel-out/local-fastbuild/genfiles/external/bazel_tools' -isystem '${execroot}/external/pcre' -isystem '${execroot}/bazel-out/local-fastbuild/genfiles/external/pcre' -isystem '${execroot}/external/bazel_tools/tools/cpp/gcc3')
ERROR: /project/pcre/BUILD.bazel:3:1: CGoCodeGen //pcre:go_default_library.cgo_codegen failed (Exit 1).
/root/.cache/bazel/_bazel_root/2ca1f4ebdc59348ffdc31d97a51a98d5/bazel-sandbox/8921902521093490301/execroot/test/pcre/pcre.go:59:18: fatal error: pcre.h: No such file or directory
 #include "pcre.h"
                  ^
compilation terminated.
CgoCodegen: error running cgo: exit status 1
INFO: Elapsed time: 4.357s, Critical Path: 1.54s
```

--- a/go/private/cgo.bzl
+++ b/go/private/cgo.bzl
@@ -76,11 +76,11 @@ def _cgo_codegen_impl(ctx):
     deps += d.cc.libs
     copts += ['-D' + define for define in d.cc.defines]
     for inc in d.cc.include_directories:
-      copts += ['-I', _exec_path(inc)]
+      copts += ['-I', inc]
     for inc in d.cc.quote_include_directories:
-      copts += ['-iquote', _exec_path(inc)]
+      copts += ['-iquote', inc]
     for inc in d.cc.system_include_directories:
-      copts += ['-isystem', _exec_path(inc)]
+      copts += ['-isystem', inc]
     for lib in d.cc.libs:
       if lib.basename.startswith('lib') and lib.basename.endswith('.so'):
         linkopts += ['-L', lib.dirname, '-l', lib.basename[3:-3]]
@@ -229,12 +229,6 @@ _cgo_object = rule(
     },
     fragments = ["cpp"],
 )
-
-
-def _exec_path(path):
-  if path.startswith('/'):
-    return path
-  return '${execroot}/' + path
 
 
 """Generates _all.o to be archived together with Go objects.

